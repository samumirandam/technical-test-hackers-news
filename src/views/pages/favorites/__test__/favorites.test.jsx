/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-unresolved */
import React from 'react';
import { render, screen, fireEvent } from '@utils/test-utils';

import Favorites from '../index';

const defaultProps = {};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<Favorites {...setupStore} />, {});
};

describe('Test for Favorites page component', () => {
  test('Should render without errors', () => {
    setup();
    expect(screen.getByTestId('Favorites')).toBeTruthy();
  });

  test('Should renders the same component', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test('Should render click in favorite icon', () => {
    localStorage.setItem(
      'FAVORITES',
      JSON.stringify([
        {
          created_at: '2002-08-26T00:27:26.000Z',
          author: 'didericis',
          story_title:
            'Threats of Blackouts Drive Japan to Embrace Nuclear Again',
          story_url:
            'https://financialpost.com/pmn/business-pmn/threats-of-blackouts-drive-japan-to-embrace-nuclear-again',
          objectID: '32601835',
        },
      ]),
    );
    setup();
    expect(localStorage.FAVORITES).toEqual(
      JSON.stringify([
        {
          created_at: '2002-08-26T00:27:26.000Z',
          author: 'didericis',
          story_title:
            'Threats of Blackouts Drive Japan to Embrace Nuclear Again',
          story_url:
            'https://financialpost.com/pmn/business-pmn/threats-of-blackouts-drive-japan-to-embrace-nuclear-again',
          objectID: '32601835',
        },
      ]),
    );
    fireEvent.click(screen.getByTestId('favorite-button'));
    expect(localStorage.FAVORITES).toEqual(JSON.stringify([]));
  });
});
