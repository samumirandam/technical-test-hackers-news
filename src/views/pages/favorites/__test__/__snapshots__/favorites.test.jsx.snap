// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Favorites page component Should renders the same component 1`] = `
<div>
  <section
    class="Favorites"
    data-testid="Favorites"
  >
    <div
      class="PageTabs Favorites__tabs"
    >
      <button
        class="Button PageTabs__tab active"
        type="button"
      >
        All
      </button>
      <button
        class="Button PageTabs__tab"
        type="button"
      >
        My faves
      </button>
    </div>
    <section
      class="StoryList"
    />
  </section>
</div>
`;
